### DISCUSSION TITLE
Check failed: b < len (21 vs. 21) slicing with begin[0]=21 exceends limit of 21

### DISCUSSION THREAD

**Discussion by bobpoekert** [apache/incubator-mxnet#13520] [Q&A]
It looks like I’m hitting this condition: https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/matrix_op-inl.h#L670

Should that ever happen or does that indicate a bug in mxnet? It’s happening consistently but I haven’t narrowed it down to a small test case.

It is a little suspicious that line 664 uses CHECK_GE but line 670 uses CHECK_LT and not CHECK_LE

Here’s the full error message: mxnet.base.MXNetError: [15:30:34] src/operator/tensor/./matrix_op-inl.h:670: Check failed: b < len (21 vs. 21) slicing with begin[0]=21 exceends limit of 21

**Comment by lanking520** [comment]
hi @bobpoekert could you please bring a minimal example that can reproduce this issue? Which MXNet version you are using? Are you using Python?

**Comment by vdantu** [comment]
@bobpoekert : Could you please provide us more information about the setup and minimum reproducible script?

**Comment by vrakesh** [comment]
@bobpoekert Has the issue been resolved for you? If not requesting you to provide us with setup and minimum reproducible script

**Comment by leleamol** [comment]
@mxnet-label-bot can this question be closed?

**Comment by vdantu** [comment]
@rahul003 : Could you help with this?

**Comment by zhaotun** [comment]
I get the same error when train my own dataset. I use VOCLike to load my dataset which described at 
[https://gluon-cv.mxnet.io/build/examples_datasets/detection_custom.html](url)，and i change the CLASSES to my own classnames ,such as CLASSES=['1','2','3', ... ,'32']，i have 32 classes and the classname is  from 1 to 32, then start the train , get the error 'Check failed: b < len (25 vs. 20) slicing with begin[3]=25 exceends limit of 20' , after many tries ,I found that the CLASSES in "C:\ProgramData\Anaconda3\Lib\site-packages\gluoncv\data\pascal_voc\detection.py" also need to be changed, when i changed it to my own classnames , the error disappeared . holp to be helpful.

**Comment by kaiserdan** [comment]
Thanks for the suggestion @zhaotun. I faced the same problem and your proposed fix solved it.
